util/mkdocstrings_patch_type_aliases/mkdocstrings_patch_type_aliases/_plugin.py

```diff
@@ -0,0 +1,76 @@
+import re
+from re import Pattern
+from typing import Callable
+
+from mkdocs.plugins import BasePlugin
+from mkdocs.structure.nav import Page
+
+ReplaceFunc = Callable[[str], str]
+
+
+def simple_replace(look_for: str, replace_with: str) -> ReplaceFunc:
+    def _replace(text: str) -> str:
+        return text.replace(look_for, replace_with)
+
+    return _replace
+
+
+def regex_replace(look_for: Pattern, replace_with: str) -> ReplaceFunc:
+    def _replace(text: str) -> str:
+        # breakpoint()
+        return re.sub(look_for, replace_with, text)
+
+    return _replace
+
+
+# Order is significant
+REPLACE_FUNCS = [
```

Comment:
wow so much work to get this in place

Reply:
Yeah. The linked issue has more detail about the type alias thing. There is also a change coming in 3.10 that would effect this, so it is extra complicated. This custom plugin was the quick and dirty fix for right now.